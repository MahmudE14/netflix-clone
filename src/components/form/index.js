import { Container, Base, Error, Title, Text, TextSmall, Link, Input, Submit } from './styles/form'

export default function Form({ children, ...restProps }) {
    return (
        <Container {...restProps}>{children}</Container>
    )
};

Form.Base = function FormContainer({ children, ...restProps }) {
    return <Base { ...restProps }>{children}</Base>
}

Form.Error = function FormContainer({ children, ...restProps }) {
    return <Error { ...restProps }>{children}</Error>
}

Form.Title = function FormContainer({ children, ...restProps }) {
    return <Title { ...restProps }>{children}</Title>
}

Form.Text = function FormContainer({ children, ...restProps }) {
    return <Text { ...restProps }>{children}</Text>
}

Form.TextSmall = function FormContainer({ children, ...restProps }) {
    return <TextSmall { ...restProps }>{children}</TextSmall>
}

Form.Link = function FormContainer({ children, ...restProps }) {
    return <Link { ...restProps }>{children}</Link>
}

Form.Input = function FormContainer({ children, ...restProps }) {
    return <Input { ...restProps }>{children}</Input>
}

Form.Submit = function FormContainer({ children, ...restProps }) {
    return <Submit { ...restProps }>{children}</Submit>
}
