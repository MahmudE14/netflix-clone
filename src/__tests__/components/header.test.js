import { fireEvent, render, screen } from "@testing-library/react";
import { Header } from "../../components";

jest.mock("react-router-dom");

describe("<Header />", () => {
  it("renders the <Header /> with a background", () => {
    const { container } = render(
      <Header>
        <Header.Frame>
          <Header.Logo src="/logo.svg" alt="Netflix" />
          <Header.TextLink active="true">Hello I am a link!</Header.TextLink>
        </Header.Frame>
      </Header>
    );

    expect(screen.getByText("Hello I am a link!")).toBeTruthy();
    expect(screen.getByTestId("header-bg")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Header /> without a background -", () => {
    const { container } = render(
      <Header bg={false}>
        <Header.Frame>
          <Header.Logo src="/logo.svg" alt="Netflix" />
          <Header.ButtonLink>Sign In</Header.ButtonLink>
          <Header.TextLink active="false">Hello I am a link!</Header.TextLink>
        </Header.Frame>
      </Header>
    );

    expect(screen.getByText("Hello I am a link!")).toBeTruthy();
    expect(screen.queryByTestId("header-bg")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the full <Header /> with a background", () => {
    const { container } = render(
      <Header src="joker1" dontShowOnSmallViewPort>
        <Header.Frame>
          <Header.Group>
            <Header.Logo src='/images/logo.svg' alt="Netflix" />
            <Header.TextLink
              onClick={() => {}}
              active={false}
            >
              Films
            </Header.TextLink>
            <Header.TextLink
              onClick={() => {}}
              active={false}
            >
              Series
            </Header.TextLink>
          </Header.Group>
          <Header.Group>
            <Header.Profile>
              <Header.Search
                searchTerm="Joker"
                setSearchTerm={() => {}}
              />
              <Header.Picture src="/images/test.png" />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src="/images/test.png" />
                  <Header.TextLink>Usama Mahmud</Header.TextLink>
                </Header.Group>
                <Header.Group>
                  <Header.TextLink onClick={() => {}}>
                    Sign Out
                  </Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>
        <Header.Feature>
          <Header.FeatureCallout>Watch Joker Now</Header.FeatureCallout>
          <Header.Text>
            Forever alone in a crowd, failed comedian Arthur Fleck seeks
            connection as he walks the streets of Gotham City. Arthur wears two
            masks -- the one he paints for his day job as a clown, and the guise
            he projects in a futile attempt to feel like he's part of the world
            around him.
          </Header.Text>
          <Header.PlayButton>Play</Header.PlayButton>
        </Header.Feature>
      </Header>
    );

    const movieDescriptionText = `Forever alone in a crowd, failed comedian Arthur Fleck seeks connection as he walks the streets of Gotham City. Arthur wears two masks -- the one he paints for his day job as a clown, and the guise he projects in a futile attempt to feel like he's part of the world around him.`

    expect(screen.getByTestId("search-input")).toBeTruthy();
    expect(screen.getByTestId("search-input").value).toBe('Joker')
    fireEvent.change(screen.getByTestId("search-input"), {
        target: { value: 'family guy' },
    })
    fireEvent.click(screen.getByTestId("search-click"))
    
    expect(screen.getByText("Series")).toBeTruthy();
    expect(screen.getByText("Films")).toBeTruthy();
    expect(screen.getByText("Usama Mahmud")).toBeTruthy()
    expect(screen.getByText("Watch Joker Now")).toBeTruthy()
    expect(screen.getByText("Sign Out")).toBeTruthy()
    expect(screen.getByText(movieDescriptionText)).toBeTruthy()
    expect(screen.getByText("Play")).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot();
  });
});
