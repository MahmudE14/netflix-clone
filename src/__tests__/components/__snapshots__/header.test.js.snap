// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders the <Header /> with a background 1`] = `
<div
  class="header__Background-sc-t8lkk2-0 fmrKPc"
  data-testid="header-bg"
>
  <div
    class="header__Container-sc-t8lkk2-1 bEDcdp"
  >
    <p
      class="header__Link-sc-t8lkk2-2 dhVuwR"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header /> renders the <Header /> without a background - 1`] = `
<div
  class="header__Container-sc-t8lkk2-1 bEDcdp"
>
  <p
    class="header__Link-sc-t8lkk2-2 ehqIHZ"
  >
    Hello I am a link!
  </p>
</div>
`;

exports[`<Header /> renders the full <Header /> with a background 1`] = `
<div
  class="header__Background-sc-t8lkk2-0 hQpTSC"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-sc-t8lkk2-1 bEDcdp"
  >
    <div
      class="header__Group-sc-t8lkk2-3 iHmbZU"
    >
      <p
        class="header__Link-sc-t8lkk2-2 ehqIHZ"
      >
        Films
      </p>
      <p
        class="header__Link-sc-t8lkk2-2 ehqIHZ"
      >
        Series
      </p>
    </div>
    <div
      class="header__Group-sc-t8lkk2-3 iHmbZU"
    >
      <div
        class="header__Profile-sc-t8lkk2-10 eyfxCQ"
      >
        <div
          class="header__Search-sc-t8lkk2-5 jldyvg"
        >
          <button
            class="header__SearchIcon-sc-t8lkk2-6 gSYwqV"
            data-testid="search-click"
          >
            <img
              alt="search"
              src="/images/icons/search.png"
            />
          </button>
          <input
            class="header__SearchInput-sc-t8lkk2-4 ctCfef"
            data-testid="search-input"
            placeholder="Search films and series"
            value="Joker"
          />
        </div>
        <button
          class="header__Picture-sc-t8lkk2-8 kzsGzO"
          src="/images/users//images/test.png.png"
        />
        <div
          class="header__Dropdown-sc-t8lkk2-9 cGILoT"
        >
          <div
            class="header__Group-sc-t8lkk2-3 iHmbZU"
          >
            <button
              class="header__Picture-sc-t8lkk2-8 kzsGzO"
              src="/images/users//images/test.png.png"
            />
            <p
              class="header__Link-sc-t8lkk2-2 ehqIHZ"
            >
              Usama Mahmud
            </p>
          </div>
          <div
            class="header__Group-sc-t8lkk2-3 iHmbZU"
          >
            <p
              class="header__Link-sc-t8lkk2-2 ehqIHZ"
            >
              Sign Out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-sc-t8lkk2-1 header__Feature-sc-t8lkk2-11 bEDcdp cDcyFR"
  >
    <h2
      class="header__FeatureCallout-sc-t8lkk2-16 eSFNff"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-sc-t8lkk2-15 hPhqIV"
    >
      Forever alone in a crowd, failed comedian Arthur Fleck seeks connection as he walks the streets of Gotham City. Arthur wears two masks -- the one he paints for his day job as a clown, and the guise he projects in a futile attempt to feel like he's part of the world around him.
    </p>
    <button
      class="header__PlayButton-sc-t8lkk2-14 eXvyBX"
    >
      Play
    </button>
  </div>
</div>
`;
